### PR TITLE
fix: change navbar dropdown icon to InitialIcon

### PR DESCRIPTION
**Pull request by elanjm12**
Change navbar dropdown icon to InitialIcon. Add text description of which user is logged in.

Screenshot from 2022-05-10 12-52-52

### CODE REVIEW

File: src/js/nav/components/NavBar.js [hunk 1]
```diff
@@ -1,9 +1,18 @@
 import { MenuButton } from "@reach/menu-button";
-import React from "react";
+import React, { useMemo } from "react";
```
Comment (ReeceHoffmann) [inline src/js/nav/components/NavBar.js]: Remove useMemo

File: src/js/nav/components/NavBar.js [hunk 2]
```diff
@@ -74,11 +83,16 @@ export const Bar = ({ administrator, dev, userId, onLogout }) => (
 
             <Dropdown>
                 <NavDropdownButton>
-                    <Icon name="user" />
+                    <InitialIcon handle={handle} size="md"></InitialIcon>
```
Comment (ReeceHoffmann) [inline src/js/nav/components/NavBar.js]:
```suggestion
                    <InitialIcon handle={handle} size="md"/>
```

File: src/js/nav/components/NavBar.js [hunk 3]
```diff
@@ -74,11 +83,16 @@ export const Bar = ({ administrator, dev, userId, onLogout }) => (
...
+
                     <span>{userId}</span>
                     <Icon name="caret-down" />
                 </NavDropdownButton>
                 <DropdownMenuList>
+                    <DropdownMenuLink to="/account">
+                        Signed in as <b>{handle}</b>
```
Comment (ReeceHoffmann) [inline src/js/nav/components/NavBar.js]:
```suggestion
                        Signed in as <strong>{handle}</strong>
```

Both should act identically in standard browsers, but can behave differently with assistive technology and non-standard browsers as `<b>` is strictly styling while `<strong>` indicates semantic importance

File: src/js/nav/components/NavBar.js [hunk 4]
```diff
@@ -74,11 +83,16 @@ export const Bar = ({ administrator, dev, userId, onLogout }) => (
...
             <Dropdown>
                 <NavDropdownButton>
-                    <Icon name="user" />
+                    <InitialIcon handle={handle} size="md"></InitialIcon>
+
                     <span>{userId}</span>
                     <Icon name="caret-down" />
                 </NavDropdownButton>
                 <DropdownMenuList>
+                    <DropdownMenuLink to="/account">
```
Comment (ReeceHoffmann) [inline src/js/nav/components/NavBar.js:89]: I think having the username act like a link to account is a good idea, but I am personally not a huge fan of how `<DropdownMenuLink>` will lightly shade the background when hovered for this exact case. You might be able to override this behavior by restyling it via styled components, or implement the link another way. (Also possible Ian will disagree with me on the shading)

File: src/js/nav/components/NavBar.js [hunk 5]
```diff
@@ -98,7 +112,8 @@ export const Bar = ({ administrator, dev, userId, onLogout }) => (
 export const mapStateToProps = state => ({
     ...state.account,
     dev: state.app.dev,
-    pending: state.app.pending
+    pending: state.app.pending,
+    handle: state.account.handle
```
Comment (ReeceHoffmann) [inline src/js/nav/components/NavBar.js]: When available, I try to use selectors to get data from redux state as they should be updated if the format of the redux state is ever changed. In this case there is the `getAccountHandle` selector from `account/selectors.js` I would use.

**Review by ReeceHoffmann — CHANGES_REQUESTED** [review]
Overall I think the visual aspects of the PR look good. (though Ian's opinion is what really matters) As a general comment you should always take a look at what changes deep source is suggesting. For this PR deep source is highlighting a couple of issues that should be addressed. (redundant imports and should use self closing tag for `initialIcon`)

Anyway, I also left some specific feedback on parts of the code I would change. Ian might disagree on the visual style notes, so I wouldn't blame you for waiting to implement those. Otherwise you should be safe to implement everything else.

**Review by igboyes — CHANGES_REQUESTED** [review]
Two things:

1. Extend thick grey bar to entire width of dropdown.
2. Commit message needs to change.
    This will appear in release notes and `InitialIcon` has no meaning to a user.
    How about: _fix: use user profile image for account drop down_

**Review by igboyes — APPROVED** [review]

**Pull request merged**